sklearn/base.py

```diff
@@ -319,7 +319,15 @@ def __getstate__(self):
             state = self.__dict__.copy()
 
         if type(self).__module__.startswith("sklearn."):
-            return dict(state.items(), _sklearn_version=__version__)
+            if "_sklearn_pickle_version" in state.keys():
+                pickle_version = state["_sklearn_pickle_version"]
+            else:
+                pickle_version = __version__
```

Comment:
I think this can be simplified to:

```suggestion
            pickle_version = state.get("_sklearn_pickle_version", __version__)
```

But as @thomasjpfan suggested, I would rather move this logic to `__set_state__` to only use a single attribute with an explicit name.

Reply:
Thanks for the feedback, I've added a new commit to the PR with your suggestions